lldb/tools/lldb-dap/DAP.cpp

```diff
@@ -856,7 +856,11 @@ lldb::SBError DAP::Disconnect(bool terminateDebuggee) {
 }
 
 llvm::Error DAP::Loop() {
-  auto cleanup = llvm::make_scope_exit([this]() { StopEventHandlers(); });
+  auto cleanup = llvm::make_scope_exit([this]() {
+    if (output.descriptor)
+      output.descriptor->Close();
```

Comment:
I'm a little worried about the lack of abstraction here. Should `{Input,Output}Stream` be a `class` instead of a `struct` and encapsulate this operation? We close the descriptor (if it's owned) when we destroy the `IOObjectSP`. So should we just reset the `IOObjectSP` pointer instead?

Reply:
I think I'd prefer to get rid of the `InputStream` and `OutputStream` classes altogether. They seem to add very little on top of `IOObjectSP` so maybe we can just store those directly, or alternative, make the relation an "is a" instead of a "has a".

Reply:
I have a patch to do just that in https://github.com/ashgti/llvm-project/commit/a57a16f3a88e958823335d839ead0043d4afc602 but I was waiting to do that until after I submitted #129155

Reply:
I also wasn't sure if I should submit the logging refactor because of https://github.com/llvm/llvm-project/pull/129294#discussion_r1977750894 so that is another PR I was waiting on figuring out first.

Reply:
Ah fair enough, let me take another look at that PR.